File: app/src/main/java/com/botijasoftware/utils/TextureRegion.java
package com.botijasoftware.utils;


public class TextureRegion extends Texture {

    private Texture original;

    public TextureRegion(Texture texture, TextureCoords coords) {
        super(texture.getID(), texture.width, texture.height);
        textCoords = coords;
        this.original = texture;
    }

    @Override
    public int getID() {
        return original.getID();
    }

    @Override
    public TextureRegion clone() {
        return new TextureRegion(original, textCoords);
    }

}

File: app/src/main/java/com/botijasoftware/utils/TextureRegion.kt
package com.botijasoftware.utils


class TextureRegion(private val original: Texture, coords: TextureCoords) : Texture(original.id, original.width, original.height) {

    init {
        textureCoords = coords
    }

    override var id: Int
        get() = original.id
        set(value: Int) {
            super.id = value
        }

    fun clone(): TextureRegion {
        return TextureRegion(original, textureCoords)
    }

}
